v0/destinations/marketo/networkResponseHandler.js

```diff
@@ -0,0 +1,251 @@
+          })
+          .build();
+      }
+      // default failure cases (keeping retryable for now)
+      throw new ErrorBuilder()
+        .setStatus(500)
+        .setMessage(
+          `Request Failed for Marketo, ${errors[0].message} (Retryable).${sourceMessage}`
+        )
+        .setDestinationResponse({ ...trimmedResponse, success: false })
+        .setStatTags({
+          destination: DESTINATION,
+          stage,
+          scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+          meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.META.RETRYABLE
+        })
+        .build();
+    }
+    // http success but data not present
+    throw new ErrorBuilder()
+      .setStatus(500)
+      .setMessage(`Request Failed for Marketo (Retryable).${sourceMessage}`)
+      .setDestinationResponse({
+        ...trimmedResponse,
+        success: false
+      })
+      .setStatTags({
+        destination: DESTINATION,
+        stage,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+        meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.META.RETRYABLE
+      })
+      .build();
+  }
+  // http failure cases
+  const { response } = clientResponse.response;
+  if (!response && clientResponse.response && clientResponse.response.code) {
+    const nodeSysErr = nodeSysErrorToStatus(clientResponse.response.code);
+    throw new ErrorBuilder()
+      .setStatus(nodeSysErr.status || 500)
+      .setMessage(`Error occured ${sourceMessage} Error: ${nodeSysErr.message}`)
+      .setStatTags({
+        destination: DESTINATION,
+        stage,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+        meta: getDynamicMeta(nodeSysErr.status || 500)
+      })
+      .build();
+  } else {
+    const temp = trimResponse(clientResponse.response);
+    throw new ErrorBuilder()
+      .setStatus(temp.status || 500)
+      .setMessage(`Error occured ${sourceMessage} Error: ${temp.statusText}`)
+      .setDestinationResponse({ ...temp, success: false })
+      .setStatTags({
+        destination: DESTINATION,
+        stage,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+        meta: getDynamicMeta(temp.status || 500)
+      })
+      .build();
+  }
+};
+
+const sendGetRequest = async (url, options) => {
+  let clientResponse;
+  try {
+    const response = await httpGET(url, options);
+    clientResponse = { success: true, response };
+  } catch (err) {
+    clientResponse = { success: false, response: err };
+  }
+  return clientResponse;
+};
+
+const sendPostRequest = async (url, data, options) => {
+  let clientResponse;
+  try {
+    const response = await httpPOST(url, data, options);
+    clientResponse = { success: true, response };
+  } catch (err) {
+    clientResponse = { success: false, response: err };
+  }
+  return clientResponse;
+};
+
+const responseTransform = destResponse => {
+  let respBody;
+  try {
+    respBody = JSON.parse(destResponse.Body);
```

Comment:
I think we should use a lowercase starting character for properties names in an Object. Can we follow that convention here ?

Reply:
We are actually getting destResponse as the string representation of a rudder-server structure where we follow the convention of having upper-case properties

Reply:
Why don't we do a `json.Marshal(destinationResponse)` & send it to Transformer ?
Also for the type of destinationResponse we can define the json while defining the struct itself
For example

```golang

type DestinationResponseT struct {
 Body string `json:"body"`
 .
 .
 .
}
```

Reply:
This is handled